proposals/0282-record-dot-syntax.rst

```diff
@@ -68,15 +69,39 @@ For the specification we focus on the changes to the parsing rules, and
 the desugaring, with the belief the type checking and renamer changes
 required are an unambiguous consequences of those.
 
-2.1 ``RecordDotSyntax`` language extension
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+2.1 Language extensions
+~~~~~~~~~~~~~~~~~~~~~~~
 
-This change adds a new language extension ``RecordDotSyntax``.
+This change adds new language extensions ``OverloadedRecordDot`` and
+``OverloadedRecordUpdate``.
+
+The effect of ``OverloadedRecordDot`` is to:
+  - allow ``.`` in record expressions;
```

Comment:
I'm troubled about this way of specifying it, because we don't *really* allow dot in updates unless `OverloadedRecordUpdate` is on too.

How about this?

* If `OverloadedRecordDot` is on:
  * The expression `e.lbl` means `getField @"lbl" e`
  * The expression `.lbl` means `getField @"lbl"`
  
  If `OverloadedRecordDot` is not on, these expressions are parsed as uses of the function `(.)`

* If `OverloadedRecordUpdate` is on:
  * The expression `e { lbl = val }` means `setField @"lbl" e val`
  
  If `OverloadedRecordUpdate` is not on, this expression means just what it does in Haskell 98.

* If `OverloadedRecordDot` and `OverloadedRecordUpdate` are on:
  * The expression `e { lbl1.lbl2 = ef }` means `e{ lbl1 = (e.lbl1){lbl2 = val} }`
  
  If either extension is not on, the expression is rejected.

Reply:
- Replaced the earlier text with your suggestion.